packages/suite-desktop/src-electron/libs/processes/TorProcess.ts

```diff
@@ -32,7 +54,37 @@ class TorProcess extends BaseProcess {
     }
 
     async start(): Promise<void> {
-        await super.start(['-f', 'torrc']);
+        const controlAuthCookiePath = path.join(this.authFilePath, 'control_auth_cookie');
+        await super.start([
+            // Try to write to disk less frequently than we would otherwise.
+            '--AvoidDiskWrites',
+            '1',
+            // Send all messages between minSeverity and maxSeverity to the standard output stream.
+            'Log',
+            'notice stdout',
+            // It should treat a startup event as cancelling any previous dormant state.
+            // use this option with caution: it should only be used if Tor is being started because
+            // of something that the user did, and not if Tor is being automatically started in the background.
+            '--DormantCanceledByStartup',
+            '1',
+            // Open this port to listen for connections from SOCKS-speaking applications.
+            '--SocksPort',
+            `${this.port}`,
+            // The port on which Tor will listen for local connections from Tor controller applications.
+            '--ControlPort',
+            `${this.controlPort}`,
+            // Setting CookieAuthentication will make Tor write an authentication cookie.
+            '--CookieAuthentication',
+            '1',
+            // If the 'CookieAuthentication' option is true, Tor writes a "magic
+            // cookie" file named "control_auth_cookie" into its data directory (or
+            // to another file specified in the 'CookieAuthFile' option)
+            // To authenticate, the controller must demonstrate that it can read the
+            // contents of the cookie file:
+            '--CookieAuthFile',
+            `${controlAuthCookiePath}`,
+        ]);
+        return this.torController.waitUntilAlive();
```

Comment:
Because the returned promise is awaited where the `start` is called.

Reply:
There are few await on the top so the whole function is wrapped as a promise, but no only that.

The expression `return this.torController.waitUntilAlive();` is returning a promise that comes from `waitUntilAlive()`.
The function `waitUntilAlive()` does not have `async` because it is returning the promise but it is not awaiting inside it. But even it does not have the `async` at the definition of the function it is still an asynchronous function because it returns a promise.

You could do as below but then you have a variable that is not really necessary:

```javascript
const itIsAlive = await this.torController.waitUntilAlive();
return itIsAlive;
```

And awaiting in the return is not recommended also:

```javascript 
return await this.torController.waitUntilAlive();
```

The whole async/await syntactic sugar is confusing very often, and we are not talking about error handling where async/await really sucks in comparison with `Promise().then().catch()`. Even callbacks are better for error handling in my opinion :P